### ISSUE TITLE
Exposed methods, crashes, on OSX

### ISSUE THREAD

**Issue by djberg96** [djberg96/sys-cpu#16]
There seem to be a couple methods on OSX that shouldn't be public, like `getloadavg` and `processor_info`.

In addition the `state` and `fpu_type` methods crash on Catalina

```
> Sys::CPU.methods(false)
=> [:num_cpu, :state, :machine, :model, :freq, :load_avg, :getloadavg, :processor_info, :sysconf, :fpu_type, :architecture]
```

**Comment by djberg96** [comment]
It seems to boil down to the fact that there's a different `processor_info` function for Darwin than for Solaris, and that's causing it to bomb.

http://web.mit.edu/darwin/src/modules/xnu/osfmk/man/processor_info.html

**Comment by djberg96** [comment]
Fixed with https://github.com/djberg96/sys-cpu/pull/17, released in 1.0.3.
